modules/tinymce/src/core/main/ts/fmt/MatchFormat.ts

```diff
@@ -67,65 +67,65 @@ const matchName = (dom: DOMUtils, node: Node, format) => {
   }
 };
 
-const matchItems = (dom: DOMUtils, node: Node, format, itemName: string, similar: boolean, vars: FormatVars) => {
-  let key, value;
+const matchItems = (dom: DOMUtils, node: Node, format: Format, itemName: string, similar: boolean, vars: FormatVars): boolean => {
   const items = format[itemName];
-  let i;
 
   // Custom match
   if (format.onmatch) {
-    return format.onmatch(node, format, itemName);
+    // onmatch is generic in a way that we can't really express without casting
+    return format.onmatch(node, format as never, itemName);
   }
 
   // Check all items
   if (items) {
     // Non indexed object
     if (typeof items.length === 'undefined') {
-      for (key in items) {
+      for (const key in items) {
         if (Obj.has(items, key)) {
-          if (itemName === 'attributes') {
-            value = dom.getAttrib(node, key);
-          } else {
-            value = FormatUtils.getStyle(dom, node, key);
+          const value = itemName === 'attributes' ? dom.getAttrib(node, key) : FormatUtils.getStyle(dom, node, key);
+          const expectedValue = FormatUtils.replaceVars(items[key], vars);
+
+          if (Type.isNullable(value) && Type.isNullable(expectedValue)) {
+            continue;
           }
 
           if (similar && !value && !format.exact) {
-            return;
+            return false;
```

Comment:
Yeah, I used to be pretty obsessed about size. So true/false was usually falsy things that doesn't work well with TS.